### PR TITLE
bugfix: emit as extension method if member of implicit class with val in constructor

### PR DESCRIPTION
**Pull request by kasiaMarek**
connected to: https://github.com/scalameta/metals/pull/6156#issuecomment-1973407601

### CODE REVIEW

File: mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala [hunk 1]
```diff
@@ -290,12 +290,27 @@ class ScalaToplevelMtags(
...
-              emitTerm(currRegion)
+              emitTerm(currRegion, isImplicit)
             }
           } else scanner.nextToken()
-          loop(indent, isAfterNewline = false, currRegion, newExpectIgnoreBody)
+          loop(
+            indent,
+            isAfterNewline = false,
+            currRegion,
+            if (isInParen) expectTemplate else newExpectIgnoreBody
```
Comment (jkciesluk) [inline mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala:312]: Why this change?

Comment (kasiaMarek) [reply mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala:312]:
```
implicit class A(val v: Int) {

}
```
So when we are on `class` we'll add something like expect implicit class body, and then in parenthesis we'll emit `val v` but we want to keep the information about the "expect implicit class body", it will be needed to construct correct region on `{`.

I wanted to write that this is safe, since there are no `{ }` inside the constructor definition, but I'm second guessing this because of `RefinedTypes`.

Comment (kasiaMarek) [reply mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala]: Fixed for refined types.

File: tests/unit/src/test/scala/tests/ScalaToplevelSuite.scala [hunk 2]
```diff
@@ -699,12 +708,14 @@ class ScalaToplevelSuite extends BaseSuite {
             val includeMembers = mode == All
             val (doc, overrides) =
               Mtags.indexWithOverrides(input, dialect, includeMembers)
-            additionalSymbolCheck(doc.symbols)
-            val symbols = doc.symbols.map(_.symbol).toList
+            // additionalSymbolCheck(doc.symbols)
+            // val symbols = doc.symbols.map(_.symbol).toList
```
Comment (jkciesluk) [inline tests/unit/src/test/scala/tests/ScalaToplevelSuite.scala:712]: Leftover?

**Review by jkciesluk — APPROVED** [review]
LGTM
